test/RSocketClientServerTest.cpp

```diff
@@ -36,15 +37,16 @@ TEST(RSocketClientServer, ConnectManySync) {
 TEST(RSocketClientServer, ConnectManyAsync) {
   auto server = makeServer(std::make_shared<HelloStreamRequestHandler>());
 
-  size_t count = 1000;
-  std::vector<std::unique_ptr<folly::ScopedEventBaseThread>> workers(count);
+  size_t count = 100;
+  size_t workerCount = 10;
+  std::vector<folly::ScopedEventBaseThread> workers(workerCount);
   std::vector<folly::Future<std::shared_ptr<RSocketClient>>> clients;
 
   std::atomic<int> executed{0};
   for (size_t i = 0; i < count; ++i) {
-    workers[i] = std::make_unique<folly::ScopedEventBaseThread>();
+    int workerId = folly::Random::rand32(workerCount);
```

Comment:
it'll only use about 63% of the scoped event bases for moderatly large count (probability of selecting every id exactly once is pretty slim)
probably not a good idea to have nondeterminism in tests in general, too